planner/core/preprocess.go

```diff
@@ -101,6 +101,10 @@ func Preprocess(ctx sessionctx.Context, node ast.Node, preprocessOpt ...Preproce
 		v.PreprocessorReturn = &PreprocessorReturn{}
 	}
 	node.Accept(&v)
+	readTS := ctx.GetSessionVars().TxnReadTS.UseTxnReadTS()
+	if readTS > 0 {
```

Comment:
Will this cause the information scheme not to match with the readTS?

Reply:
After we execute `set transaction read only as of timestamp ...`, the infoschema will be stored in `vars.SnapshotInfoschema`

Reply:
Also we have a test `TestSetTransactionInfoSchema` to ensure infoschema

Reply:
There are many places where inforschema is used.  In there, `PreprocessorReturn` will bring the schema, and the set will also `vars.SnapshotInfoschema` store the schema.  maybe we need to sort it in the future, and it's better to have a unified place.